### PR TITLE
[build tools] Change Rush env variable name used for doc generation

### PR DESCRIPTION
**Pull request by Ben-Pusey-Bentley**
Changes the env variable name used to determine the project's root so that repos that are not using rush do not have to define a `RUSHSTACK_*`  variable.

### CODE REVIEW

File: common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/build-tools",
+      "comment": "Changes name of environment variable that provides project root directory",
+      "type": "none"
```
[the remark this answers is not in the document]
Comment (Ben-Pusey-Bentley) [reply common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json:6]: Alright, will do.

File: tools/build/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ The default behaviors of the scripts (without parameters) assume that the direct
   - package.json
   - tsconfig.json
 
+If the directory structure fo your package varies from this, define either RUSHSTACK_FILE_ERROR_BASE_FOLDER or NODE_PROJECT_ROOT_DIRECTORY environment variables as the path to the directory containing your package.json file.
```
Comment (ben-polinsky) [inline tools/build/README.md]: 1. "of" not "fo"
2. Suggest users use the NODE_PROJECT* env variable, and maybe note parenthetically that the RUSHSTACK_FILE_* env var is deprecated.

File: common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ben-polinsky) [reply common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json:6]: Thanks - should still be a patch or minor change. Technically, it's minor (adding "api").

File: common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-19-48.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (ben-polinsky) [inline common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-19-48.json:1]: The two changes can be combined

File: tools/build/README.md (hunk 2 above)
Comment (ben-polinsky) [reply tools/build/README.md]: Oh was looking at outdated... for not of, got it.

File: common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/build-tools",
+      "comment": "Changes name of environment variable that provides project root directory. Allows use of RUSHSTACK variable as well and updates README.",
+      "type": "minor"
```
Comment (aruniverse) [inline common/changes/@itwin/build-tools/BenPusey-change-rush-env-variable-name_2023-08-09-18-41.json:6]: everything is in lockstep in this repo , so this makes no difference

```suggestion
      "type": "none"
```

File: tools/build/scripts/utils/addPackageMetadata.js [hunk 5]
```diff
@@ -7,8 +7,10 @@ const FS = require("fs-extra");
 const path = require("path");
 
 // We cannot guarantee the folder structure of a project
-// so find the project root using rush env variable if available.
-const rootPackageJson = process.env.RUSHSTACK_FILE_ERROR_BASE_FOLDER
+// so find the project root using environment variables if available, starting with NODE_PROJECT, then RUSHSTACK.
+const rootPackageJson = process.env.NODE_PROJECT_ROOT_DIRECTORY
```
Comment (aruniverse) [inline tools/build/scripts/utils/addPackageMetadata.js:11]: so basically ?

```ts
path.join(process.env.NODE_PROJECT_ROOT_DIRECTORY || process.env.RUSHSTACK_FILE_ERROR_BASE_FOLDER || "../../../../", "package.json")
```

**Review by aruniverse — APPROVED** [review]

**Review by ben-polinsky — APPROVED** [review]

**Pull request merged**